lib/rules/require-anonymous-functions.js

```diff
@@ -3,7 +3,11 @@
  *
  * Type: `Boolean`
  *
- * Value: `true`
+ * Values:
+ *  - `true`
+ *  - `Object`:
+ *    - `'allExcept'` array of exceptions:
+ *       - `'functionDeclarations'` ignores function declarations
```

Comment:
Since rule already has 'functions' in their name, maybe we could just use `declarations` instead of `functionDeclarations`?